### PR TITLE
Add RB projection tutorial

### PR DESCRIPTION
**Pull request by sdrave**

### CODE REVIEW

File: docs/source/tutorial_projection.rst [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+solving the FOM works. Let's take a look, what
+:meth:`~pymor.models.interface.Model.solve` actually does:
+
+.. jupyter-execute::
+
+    from pymor.tools.formatsrc import print_source
+    print_source(fom.solve)
+
+This does not look too interesting. Actually, :meth:`~pymor.models.interface.Model.solve`
+is just a thin wrapper around the `solve_` method, which performs the actual
```
Comment (HenKlei) [inline docs/source/tutorial_projection.rst:89]:
```suggestion
is just a thin wrapper around the `_solve` method, which performs the actual
```

File: docs/source/tutorial_projection.rst [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+
+    L(u(\mu); \mu) = F(\mu)
+
+Here, :math:`L` is a linear or non-linear parametric |Operator| and :math:`F` is a
+parametric right-hand side vector. In |StationaryModel|, :math:`L` is represented by
+the :attr:`~pymor.models.basic.StationaryModel.operator` attribute. So ::
+
+    self.operator.apply_inverse(X, mu=mu)
+
+determines the solution of this equation for the |parameter values| mu and a right-hand
```
Comment (HenKlei) [inline docs/source/tutorial_projection.rst:122]:
```suggestion
determines the solution of this equation for the |parameter values| `mu` and a right-hand
```

File: docs/source/tutorial_projection.rst [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+
+.. jupyter-execute::
+
+    from pymor.operators.interface import Operator
+    print_source(Operator.as_range_array)
+
+we see that all that :meth:`~pymor.operators.interface.Operator.as_range_array`
+does is to apply the operator to :math:`1`. (`NumpyMatrixOperator.as_range_array`
+has an optimized implementation which just converts the stored matrix to a
+|NumpyVectorArray|.
```
Comment (HenKlei) [inline docs/source/tutorial_projection.rst:152]:
```suggestion
|NumpyVectorArray|.)
```

File: docs/source/tutorial_projection.rst [hunk 4]
```diff
@@ -0,0 +1,664 @@
...
+Building the ROM
+----------------
+
+So far, we only constructed the ROM in the form of |NumPy| data structures:
+
+.. jupyter-execute::
+
+    type(reduced_operator)
+
+To build a proper pyMOR |Model| for the ROM, which can be used everywhere a |Model| is
```
Comment (HenKlei) [inline docs/source/tutorial_projection.rst:331]:
```suggestion
To build a proper pyMOR |Model| for the ROM, which can be used everywhere, a |Model| is
```

File: docs/source/tutorial_projection.rst [hunk 5]
```diff
@@ -0,0 +1,664 @@
...
+    tac = perf_counter()
+    rom.solve()
+    tuc = perf_counter()
+    print(f'FOM:          {toc-tic:.5f} (s)')
+    print(f'ROM assemble: {tac-tic:.5f} (s)')
+    print(f'ROM solve:    {tuc-tac:.5f} (s)')
+
+we see that we actually don't save anything in terms of computation time,
+as the assembly of the FOM (which involves a lot of full-order dimensional
+operations eats up all of the speedup of the ROM.
```
Comment (HenKlei) [inline docs/source/tutorial_projection.rst:395]:
```suggestion
operations) eats up all of the speedup of the ROM.
```

File: docs/source/tutorial_projection.rst [hunk 6]
```diff
@@ -0,0 +1,664 @@
...
+in particular is useful when assessing the ROM for different basis sizes. The
+actual projection is handled in the
+:meth:`~pymor.reductor.basic.StationaryRBReductor.project_operators` where we can
+find some well-known code:
+
+.. jupyter-execute::
+
+    print_source(reductor.project_operators)
+
+We also see that the reductor also takes care of projecting output functionals and
```
Comment (HenKlei) [inline docs/source/tutorial_projection.rst:639]:
```suggestion
We also see that the reductor takes care of projecting output functionals and
```

**Comment by sdrave** [comment]
The rendered tutorial can be found [here](https://docs.pymor.org/tutorial-projection/tutorial_projection.html)

File: docs/source/tutorial_projection.rst [hunk 7]
```diff
@@ -0,0 +1,661 @@
...
+.. jupyter-execute::
+
+    (U-U_N).norm(fom.h1_0_product) / U.norm(fom.h1_0_product)
+
+With only 10 basis vectors, we have achieved a relative :math:`H^1`-error of 2%.
+We can also visually inspect our solution and the approximation error:
+
+.. jupyter-execute::
+
+    fom.visualize((U, U_N, U-U_N), title=('FOM', 'ROM', 'Error'), separate_colorbars=True)
```
[the remark this answers is not in the document]
Comment (sdrave) [reply docs/source/tutorial_projection.rst]: Ok, I removed the titles in the call. Probably not worth the effort of supporting titles in the current implementations as we are about to change backends again ..

**Review by pmli — APPROVED** [review]

**Pull request merged**